File: wireframe/src/main/java/ru/nsu/e/shelbogashev/wireframe/wireframe/WireframeFrame.java
package ru.nsu.e.shelbogashev.wireframe.wireframe;

import ru.nsu.e.shelbogashev.wireframe.bsplineeditor.BSpline;

import javax.swing.*;
import java.awt.*;

/**
 * Представляет собой графическое окно для отображения проволочной модели.
 */
public class WireframeFrame extends JFrame {
    private final WireframePanel wireframePanel;

    /**
     * Создает новый экземпляр графического окна для проволочной модели.
     *
     * @param bSpline Кривая Безье, на основе которой будет создана проволочная модель.
     */
    public WireframeFrame(BSpline bSpline) {
        wireframePanel = new WireframePanel(bSpline);
        int width = 900;
        int height = 600;
        setMinimumSize(new Dimension(width, height));

        add(wireframePanel);

        setVisible(true);
    }

    /**
     * Создает проволочную модель и отображает ее на панели.
     */
    public void createWireframe() {
        wireframePanel.createWireframe();
    }

}

File: wireframe/src/main/java/ru/nsu/e/shelbogashev/wireframe/wireframe/WireframeFrame.kt
package ru.nsu.e.shelbogashev.wireframe.wireframe

import ru.nsu.e.shelbogashev.wireframe.bsplineeditor.BSpline
import java.awt.Dimension
import javax.swing.JFrame

/**
 * Представляет собой графическое окно для отображения проволочной модели.
 */
class WireframeFrame(bSpline: BSpline?) : JFrame() {
    private val wireframePanel = WireframePanel(bSpline!!)

    /**
     * Создает новый экземпляр графического окна для проволочной модели.
     *
     * @param bSpline Кривая Безье, на основе которой будет создана проволочная модель.
     */
    init {
        val width = 900
        val height = 600
        minimumSize = Dimension(width, height)

        add(wireframePanel)

        isVisible = true
    }

    /**
     * Создает проволочную модель и отображает ее на панели.
     */
    fun createWireframe() {
        wireframePanel.createWireframe()
    }
}
